docs/design_proposals/multi_environment_proposal.md

```diff
@@ -345,3 +346,74 @@ pixi run test
     EXPOSE 8080
     CMD ["/usr/local/bin/pixi", "run", "--env", "prod", "serve"]
     ```
+
+??? tip "Multiple machines from one project"
+    This is an example for an ML project that should be executable on a machine that supports `cuda` and `mlx`. It should also be executable on machines that don't support `cuda` or `mlx`, we use the `cpu` feature for this.
+    ```toml title="pixi.toml"
+    [project]
+    name = "my-ml-project"
+    description = "A project that does ML stuff"
+    authors = ["Your Name <your.name@gmail.com>"]
+    channels = ["conda-forge", "pytorch"]
+    platforms = [] # there is no platform that supports all features
+
+    [tasks]
+    train-model = "python train.py"
+    evaluate-model = "python test.py"
+
+    [dependencies]
+    python = "3.11.*"
+    pytorch = {version = ">=2.0.1", channel = "pytorch"}
+    torchvision = {version = ">=0.15", channel = "pytorch"}
+    polars = ">=0.20,<0.21"
+    matplotlib-base = ">=3.8.2,<3.9"
+    ipykernel = ">=6.28.0,<6.29"
+
+    [feature.cuda]
+    platforms = ["win-64", "linux-64"]
+    channels = [{name = "nvidia", priority = "-1"}, "pytorch"]
+    system-requirements = {cuda = "12.1"}
+
+    [feature.cuda.tasks]
+    # these tasks override the default tasks
+    train-model = "python train.py --cuda"
+    evaluate-model = "python test.py --cuda"
+
+    [feature.cuda.dependencies]
+    pytorch-cuda = {version = "12.1.*", channel = "pytorch"}
+
+    [feature.mlx]
+    platforms = ["osx-arm64"]
+
+    [feature.mlx.tasks]
+    # these tasks override the default tasks
```

Comment:
I don't really have strong opinions if we should allow overriding tasks or not.
Alternatively, you would do

```toml
[feature.cpu.tasks]
train-model = "python train.py"
evaluate-model = "python test.py"
```

Reply:
```suggestion
```
I would remove the mention for now.
The logic described in the document says we overwrite the tasks if an environment is given or requests the user to specify the required environment.